xugrid/regrid/gridder.py

```diff
@@ -0,0 +1,89 @@
+from typing import Callable, Union
+
+import xarray as xr
+
+import xugrid
+from xugrid.core.sparse import MatrixCSR
+from xugrid.regrid import reduce
+from xugrid.regrid.network import Network2d
+from xugrid.regrid.regridder import BaseRegridder, make_regrid, setup_grid
+from xugrid.regrid.structured import StructuredGrid2d
+from xugrid.regrid.unstructured import UnstructuredGrid2d
+
+
+def convert_to_match(source, target):
```

Comment:
I think you basically need this:

```python
def convert_to_match(source, target):
    PROMOTIONS = {
        frozenset({StructuredGrid2d}): UnstructuredGrid2d,
        frozenset({UnstructuredGrid2d}): UnstructuredGrid2d,
    }
    types = set({type(target)})
    matched_type = PROMOTIONS[frozenset(types)]
    return source, target.convert_to(matched_type)

Reply:
This will just convert the structured form to the unstructured from.

Reply:
Thanks, added this and implemented unittests for it!